src/libs/tiny_ttf/lv_tiny_ttf.c

```diff
@@ -205,31 +218,51 @@ static bool ttf_get_glyph_dsc_cb(const lv_font_t * font, lv_font_glyph_dsc_t * d
         dsc_out->is_placeholder = false;
         return true;
     }
+
     ttf_font_desc_t * dsc = (ttf_font_desc_t *)font->dsc;
-    int g1 = stbtt_FindGlyphIndex(&dsc->info, (int)unicode_letter);
-    if(g1 == 0) {
-        /* Glyph not found */
+
+    tiny_ttf_glyph_cache_data_t search_key = {
+        .unicode = unicode_letter,
+    };
+
+    lv_cache_entry_t * entry = lv_cache_acquire_or_create(dsc->glyph_cache, &search_key, (void *)dsc);
+
+    if(entry == NULL) {
+        LV_LOG_ERROR("cache not allocated");
         return false;
     }
-    int x1, y1, x2, y2;
 
-    stbtt_GetGlyphBitmapBox(&dsc->info, g1, dsc->scale, dsc->scale, &x1, &y1, &x2, &y2);
-    int g2 = 0;
-    if(unicode_letter_next != 0) {
-        g2 = stbtt_FindGlyphIndex(&dsc->info, (int)unicode_letter_next);
+    tiny_ttf_glyph_cache_data_t * data = lv_cache_entry_get_data(entry);
+    *dsc_out = data->glyph_dsc;
+
+    /*Kerning correction*/
+    if(font->kerning == LV_FONT_KERNING_NORMAL &&
+       unicode_letter_next != 0) { // check if we need to do any kerning calculations
+        uint32_t g1 = data->glyph_dsc.gid.index;
+
+        int g2 = 0;
+        search_key.unicode = unicode_letter_next; // reuse search key
+        lv_cache_entry_t * entry_next = lv_cache_acquire_or_create(dsc->glyph_cache, &search_key, (void *)dsc);
+
+        if(entry_next == NULL)
+            g2 = stbtt_FindGlyphIndex(&dsc->info, (int)unicode_letter_next);
```

Comment:
I didn't actually check what behaviour would be if cache size is zero - my assumption was that could return NULL then, as the code I started atop was doing a NULL check on acquired entry.

Reply:
Oh of course, if the cache size is zero then `lv_cache_acquire_or_create` will always fail and this line is what creates it, in that case.

You answered my question: yes it can return non-zero.

Thanks